src/shims/unix/fs.rs

```diff
@@ -1659,6 +1659,68 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             }
         }
     }
+
+    fn realpath(
+        &mut self,
+        path_op: &OpTy<'tcx, Tag>,
+        processed_path_op: &OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, Pointer<Option<Tag>>> {
+        let this = self.eval_context_mut();
+        this.assert_target_os_is_unix("realpath");
+
+        let pathname = this.read_path_from_c_str(this.read_pointer(path_op)?)?;
```

Comment:
Please start by reading all the arguments. It's important we read all arguments on all paths through a shim. E.g., if `processed_path_op` is uninit, we want that to be UB even if `result` below is `Err`.

See the other shims, they all do it like that.